server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/methods/SetMailboxesUpdateProcessor.java

```diff
@@ -266,14 +273,28 @@ private void updateMailbox(Mailbox mailbox, MailboxUpdateRequest updateRequest,
                     .toMailboxAcl(),
                 mailboxSession);
         }
+
         if (!originMailboxPath.equals(destinationMailboxPath)) {
-            mailboxManager.renameMailbox(mailbox.getId(), destinationMailboxPath, mailboxSession);
+            for (MailboxMetaData mailboxMetaData: originalMailboxPathWithChild) {
```

Comment:
:-1:

 - We are not resiliant to concurrent chanes
 - We are performing unecessary extra reads.
 - We loose subscriptions if renames fails.
 - We are building potentially unfounded expectation on the rename output.

All the above mentioned points leads to brittle potentially buggy code.

Why not make `MailboxManager::rename` return a `MailboxRenamed` object with the mailboxes that it did actually renamed?

```
List<RenamedMailbox> renamesMailboxes = mailboxManager.renameMailbox(...);

renamesMailboxes.forEach(this::updateSubscription);
```

Reply:
I agree with `potentially buggy code` i made.
I dont agree with `We are performing unecessary extra reads`.
```
 we need originalMailboxPath + childPath and destinationMailboxPath + childPath
```
As i known, no way to get `childPath` without reading from DB?I would happy with your help.

Reply:
For sure

Reply:
To rephrase it whatI propose is to have the mbx manager return the entities it actually modified. 

That way we don't guess what it does, and we don't need extra reads.

Reply:
ok, i agree with modify the mbx manager api.